### PR TITLE
Add gelf serializer & graylog output filter.

### PR DESCRIPTION
**Pull request by vanillahsu**
Dear sir:

I make 2 plugins for graylog, please help to review, thanks.

### CODE REVIEW

**Comment by sparrc** [comment]
if the gelf serializer is only used for graylog, then please don't make it a generalized serializer, just write the serialization code within the graylog output plugin.

File: Godeps_windows [hunk 1]
```diff
@@ -1,3 +1,4 @@
+github.com/Graylog2/go-gelf/gelf 76d60fc890684cff2fb65c631ed711d4af080592
```
Comment (sparrc) [inline Godeps_windows]: this is a fork of a fork of a fork of https://github.com/SocialCodeInc/go-gelf, why??

File: Godeps_windows [hunk 2]
```diff
@@ -48,6 +49,7 @@ github.com/shirou/w32 ada3ba68f000aa1b58580e45c9d308fe0b7fc5c5
 github.com/soniah/gosnmp b1b4f885b12c5dcbd021c5cee1c904110de6db7d
 github.com/streadway/amqp b4f3ceab0337f013208d31348b578d83c0064744
 github.com/stretchr/testify 1f4a1643a57e798696635ea4c126e9127adb7d3c
+github.com/vanillahsu/graylog-golang fb9318c61dfb180cf66e671c7362caa8720a048c
```
Comment (sparrc) [inline Godeps_windows]: this is also a fork....why?

**Comment by sparrc** [comment]
In general I prefer not to rely on forks when importing dependencies, because maintenance of forks is less reliable. Also please ping me when you make changes.

**Comment by vanillahsu** [comment]
Hi:

I've try github.com/Graylog2/go-gelf/gelf and mainstream of graylog-golang,

but it's not good enough for me,

go-gelf support io.Multi interface, but will handle graylog format on it's Write function,
graylog-golang do extra json encode/decode when it send to graylog.

that's why I fork graylog-golang and remove extra works.

**Comment by sparrc** [comment]
the code here is rather simple, couldn't you just copy it locally? less dependencies is better, especially when the dependency is this small: https://github.com/vanillahsu/graylog-golang/blob/master/gelf.go

**Comment by vanillahsu** [comment]
Sure, I will do it later, thanks.

**Comment by sparrc** [comment]
is this ready?

**Comment by vanillahsu** [comment]
Sure, whole gelf related function moved to graylog.go.

**Comment by sparrc** [comment]
two things still needed:

1. write a README
2. remove the Godeps_windows file change

**Comment by vanillahsu** [comment]
Hi:

I already add README.md & unit test, please help to check, thanks.

**Comment by sparrc** [comment]
@vanillahsu I merged your PR but now I think I will need to revert. Can you explain how this unit tests makes any sense?

It sends `int64(1234567890)` but expects to receive `float64(1)`? There must be something wrong in the code for this test to ever work.

```
+func TestWrite(t *testing.T) {
 +	var wg sync.WaitGroup
 +	wg.Add(1)
 +	go UDPServer(t, &wg)
 +
 +	i := Graylog{
 +		Servers: []string{"127.0.0.1:12201"},
 +	}
 +	i.Connect()
 +
 +	metrics := testutil.MockMetrics()
 +	metrics = append(metrics, testutil.TestMetric(int64(1234567890)))
 +
 +	i.Write(metrics)
 +
 +	wg.Wait()
 +	i.Close()
 +}
 +
 +type GelfObject map[string]interface{}
 +
 +func UDPServer(t *testing.T, wg *sync.WaitGroup) {
 +	serverAddr, _ := net.ResolveUDPAddr("udp", "127.0.0.1:12201")
 +	udpServer, _ := net.ListenUDP("udp", serverAddr)
 +	defer wg.Done()
 +
 +	bufR := make([]byte, 1024)
 +	n, _, _ := udpServer.ReadFromUDP(bufR)
 +
 +	b := bytes.NewReader(bufR[0:n])
 +	r, _ := zlib.NewReader(b)
 +
 +	bufW := bytes.NewBuffer(nil)
 +	io.Copy(bufW, r)
 +	r.Close()
 +
 +	var obj GelfObject
 +	json.Unmarshal(bufW.Bytes(), &obj)
 +	assert.Equal(t, obj["_value"], float64(1))
```

**Comment by vanillahsu** [comment]
Hi:

because metrics is an array, it contain float(1) and int64(1234567890), send to udp server with 2 packets.

but on UDPServer function, I checked first packet, so it's float(1), is it make sense?

**Comment by sparrc** [comment]
oh I see that now, sorry. But there is no guarantee that those will appear in-order. Why send both metrics? Why not just send the one?

Please re-push the PR with the test changed unless there's a good reason for it.

**Comment by vanillahsu** [comment]
Hi:

I remove second metric from unit test, please help to check, thanks.

**Pull request merged**